lib/matplotlib/backend_bases.py

```diff
@@ -3387,17 +3387,15 @@ def add_tools(self, tools):
         ----------
         tools : List
             List in the form
-            [[group1, [(Tool1, name1), (Tool2, name2) ...]][group2...]]
-            where group1 is the name of the group where the
-            Tool1, Tool2... are going to be added, and name1, name2... are the
-            names of the tools
+            [(Tool1, name1), (Tool2, name2) ...]
+            where Tool1, name1 represent the tool, and the respective name
+            of the tool which gets used as an id.
         """
 
-        for group, grouptools in tools:
-            for position, tool in enumerate(grouptools):
-                self.add_tool(tool[1], tool[0], group, position)
+        for tool, name in tools:
+            self.add_tool(name, tool)
 
-    def add_tool(self, name, tool, group=None, position=None):
+    def add_tool(self, name, tool, *args, **kwargs):
```

Comment:
Because passing we only want to pass classes and not objects to ``add_tool``, this means that one can't set up the tool in __init__, which seems odd to me.  The tool I have written, I have to create extra methods such as, and then when I come to add the tool do:

```python
tool = navigation.add_tool('my tool', MyTool)
tool.set_foo(foo)
tool.set_bar(bar)
```

rather than
```python
navigation.add_tool('my tool', MyTool, foo=foo, bar=bar)
```
It just feels nicer and more python and more user friendly.

I don't think this will affect tools from getting merged with master at all.  If they require other stuff set, then they wouldn't get added via ``add_tools`` anyway, and if they didn't require it, then it doesn't matter that they don't get the settings, because they were designed not to require them.